### ISSUE TITLE
How to pass test images for sentence retrieval?

### ISSUE THREAD

**Issue by architlatkar27** [clip-italian/clip-italian#5]
Hi there,
I am trying to use the colab  notebook for text retrieval using image. 
I want to pass my own set of images and text to the model. How can i do that?
Thanks in advance

**Comment by architlatkar27** [comment]
Also, what exactly does "tops" list in the code signify? 
Could you describe the outputs in the last cell?

**Comment by vinid** [comment]
Hello @architlatkar27,

I don't have access to the notebooks right now, so I might need to revise this later, but:

For question 1):

You need to upload your images in a folder on colab and then you just need to update the path used in the dataset and in the image search function.

For the text, we take mscoco json and use the text from there, you can just modify that part (this if you want to do a proper evaluation, if you are just testing image search, you can just enter queries in the method to search images)

For question 2)

Can you point me to the code where you found this?

*Edited

**Comment by architlatkar27** [comment]
![image](https://user-images.githubusercontent.com/52096846/129245171-26075d07-f79a-4d59-8d58-8c2090c4a0b7.png)
When i ran the above cell (which happens to be the last one in CLIP-italian ImageNet Zero Shot Evaluation .ipynb file) I am getting the following output:
`{'top1': 20.16, 'top5': 36.578, 'top10': 42.906, 'top100': 67.11}`

I am not exactly sure on how to interpret this result. And how to get the actual sentences instead of the values.

**Comment by vinid** [comment]
`{'top1': 20.16, 'top5': 36.578, 'top10': 42.906, 'top100': 67.11}` is the accuracy@K

the function `zeroshot_classifier` creates the embeddings for the sentences* once this is done, you can just multiply the image embeddings (`image_features`) with the `zeroshot_weights` to get the similarity scores (and you can then pick the most similar image-text pairs.

This is done with the operation `logits = 100. * image_features @ zeroshot_weights`

The code in the [demo](https://huggingface.co/spaces/clip-italian/clip-italian-demo/blob/main/image2text.py#L67) might be useful to implement this.

Let me know if this is not clear!

*note that in the code we combine templates and class names to replicate the experiment run in the OpenAI paper on CLIP. You probably do not need this and you can just have the sentences you need.
